### ISSUE TITLE
Add unique id to the web form element in order to support screen reader

### ISSUE THREAD

**Issue by fendiaoxiaoshuazi** [enketo/enketo#783]
**Description**
Our customers keep pushing us to make the Survey123 webform fully WCAG/508 compliant since this is a showstopper for them, e.g. the webform used on the government website needs to be fully WCAG/508 compliant.

We have made lots of effort to make the Survey123 keyboard navigatable, and we will need to keep going to make Survey123 support screen reader. To fully support the screen reader, most of the form elements need to have a unique id, and then we can use aria-labelledby, aira-describedby, and etc. to make the screen reader behave as expected. 

enketo already has aria-label added for elements, but it is not fully WCAG compliant, especially, we cannot fulfill the requirements in the VPAT, here is an example in the screenshot below:

![image](https://user-images.githubusercontent.com/26078106/102059051-58d80500-3e2b-11eb-99b5-e3c04e83577b.png)

Ref: Survey123 Web App VPAT
https://www.esri.com/content/dam/esrisites/en-us/media/legal/vpats/survey123-for-arcgis-311-vpat.pdf

**Expected behavior**
The unique id is needed for the label, description, input element, error msg, hint msg, etc.

Here is the demonstration for adding aria labels:

![7d7e9100-5ffe-11ea-82d1-d1cd545b19ad](https://user-images.githubusercontent.com/26078106/102059085-62fa0380-3e2b-11eb-9502-c5680fcbd52a.png)

cc @yixi8524 @zhif7782 @nanzhang01
